core/domain/story_domain.py

```diff
@@ -0,0 +1,359 @@
+        self.destination_nodes = destination_nodes
+        self.acquired_skill_ids = acquired_skill_ids
+        self.prerequisite_skill_ids = prerequisite_skill_ids
+        self.annotations = annotations
+        self.exploration_id = exploration_id
+
+    def to_dict(self):
+        """Returns a dict representing this StoryNode domain object.
+
+        Returns:
+            A dict, mapping all fields of StoryNode instance.
+        """
+        return {
+            'id': self.id,
+            'destination_nodes': self.destination_nodes,
+            'acquired_skill_ids': self.acquired_skill_ids,
+            'prerequisite_skill_ids': self.prerequisite_skill_ids,
+            'annotations': self.annotations,
+            'exploration_id': self.exploration_id
+        }
+
+    @classmethod
+    def from_dict(cls, node_dict):
+        """Return a StoryNode domain object from a dict.
+
+        Args:
+            node_dict: dict. The dict representation of StoryNode object.
+
+        Returns:
+            StoryNode. The corresponding StoryNode domain object.
+        """
+        node = cls(
+            node_dict['id'], node_dict['destination_nodes'],
+            node_dict['acquired_skill_ids'],
+            node_dict['prerequisite_skill_ids'], node_dict['annotations'],
+            node_dict['exploration_id'])
+
+        return node
+
+    @classmethod
+    def create_default_story_node(cls, node_id):
+        """Returns a StoryNode domain object with default values.
+
+        Args:
+            node_id: str. The id of the node.
+
+        Returns:
+            StoryNode. The StoryNode domain object with default
+            value.
+        """
+        return cls(node_id, [], [], [], '', '')
+
+
+class Story(object):
+    """Domain object for an Oppia Story."""
+
+    def __init__(
+            self, story_id, title, description, notes, topic,
+            story_nodes, schema_version, language_code, version,
+            created_on=None, last_updated=None):
+        """Constructs a Question domain object.
+
+        Args:
+            story_id: str. The unique ID of the story.
+            title: str. The title of the story.
+            description: str. The high level desscription of the story.
+            notes: str. A set of notes, that describe the characters,
+                main storyline, and setting.
+            topic: str. The topic id the story corresponds to.
+            story_nodes: list(StoryNode). List of the story nodes present in
+                the story.
+            created_on: datetime.datetime. Date and time when the story is
+                created.
+            last_updated: datetime.datetime. Date and time when the
+                story was last updated.
+            schema_version: int. The schema version for the story.
+            language_code: str. The ISO 639-1 code for the language this
+                story is written in.
+            version: int. The version of the story.
```

Comment:
Is this version or schema version?

If the former, I don't think it's needed in the domain object. For one thing it doesn't make sense to specify a version in from_dict....

(Edit: I noticed this was part of Exploration. Let's maybe talk about it and check if it's needed.)

Reply:
There, it was incremented each time, the exploration was updated and was used to revert back to a version at a later stage. Is that needed in the Story construct now, or would it be implemented later?

Reply:
Yep